app/api/items/ask_hint.go

```diff
@@ -0,0 +1,244 @@
+
+		var hintsRequestedNew []byte
+		hintsRequestedNew, err = json.Marshal(hintsRequestedParsed)
+		service.MustNotBeError(err)
+		hintsRequestedNewString := string(hintsRequestedNew)
+		requestData.TaskToken.HintsRequested = &hintsRequestedNewString
+		hintsGivenCountString := strconv.Itoa(len(hintsRequestedParsed))
+		requestData.TaskToken.HintsGivenCount = &hintsGivenCountString
+
+		columnsToUpdate := map[string]interface{}{
+			"sHintsRequested":            hintsRequestedNew,
+			"nbHintsCached":              len(hintsRequestedParsed),
+			"nbTasksWithHelp":            1,
+			"sAncestorsComputationState": "todo",
+			"sLastActivityDate":          gorm.Expr("NOW()"),
+			"sLastHintDate":              gorm.Expr("NOW()"),
+		}
+		// Update groups_attempts with the hint request
+		if requestData.TaskToken.Converted.AttemptID != nil {
+			service.MustNotBeError(store.GroupAttempts().ByID(*requestData.TaskToken.Converted.AttemptID).
+				UpdateColumn(columnsToUpdate).Error())
+		}
+		// Update users_items with the hint request
+		query := store.UserItems().Where("idUser = ?", user.UserID).
+			Where("idItem = ?", requestData.TaskToken.Converted.LocalItemID)
+		if requestData.TaskToken.Converted.AttemptID != nil {
+			query = query.Where("idAttemptActive = ?", *requestData.TaskToken.Converted.AttemptID)
+		}
+		service.MustNotBeError(query.UpdateColumn(columnsToUpdate).Error())
+		service.MustNotBeError(store.GroupAttempts().After())
+
+		return nil
+	})
+	if apiError != service.NoError {
+		return apiError
+	}
+	service.MustNotBeError(err)
+
+	requestData.TaskToken.PlatformName = srv.TokenConfig.PlatformName
+	newTaskToken, err := requestData.TaskToken.Sign(srv.TokenConfig.PrivateKey)
+	service.MustNotBeError(err)
+
+	service.MustNotBeError(render.Render(w, r, service.CreationSuccess(map[string]interface{}{
+		"task_token": newTaskToken,
+	})))
+	return service.NoError
+}
+
+func queryAndParsePreviouslyRequestedHints(
+	requestData *AskHintRequest, store *database.DataStore, user *database.User, r *http.Request) ([]payloads.Anything, error) {
+	fieldsForLogging := map[string]interface{}{
+		"idUser": user.UserID,
+		"idItem": requestData.TaskToken.Converted.LocalItemID,
+	}
+	var query *database.DB
+	if requestData.TaskToken.Converted.AttemptID != nil {
+		query = store.GroupAttempts().ByID(*requestData.TaskToken.Converted.AttemptID)
+		fieldsForLogging["idAttempt"] = *requestData.TaskToken.Converted.AttemptID
+	} else {
+		query = store.UserItems().Where("idUser = ?", user.UserID).
+			Where("idItem = ?", requestData.TaskToken.Converted.LocalItemID)
+	}
+	var hintsRequested *string
+	err := query.PluckFirst("sHintsRequested", &hintsRequested).Error()
+	var hintsRequestedParsed []payloads.Anything
+	if err == nil && hintsRequested != nil {
+		hintsErr := json.Unmarshal(*(*[]byte)(unsafe.Pointer(hintsRequested)), &hintsRequestedParsed) //nolint:gosec
```

Comment:
From what I understand `Unmarshal` does not guarantee the `dest` in untouched in case of error. So `hintsRequestedParsed` should be reset to an empty array in case of error I think as you "ignore" the error and continue in this case. May worth a test.

Reply:
Fixed. The test has been already there